packages/block-editor/src/hooks/style.js

```diff
@@ -85,21 +86,35 @@ export function getInlineStyles( styles = {} ) {
 			// option and backwards compatibility for border radius support.
 			const styleValue = get( styles, path );
 
-			if ( !! subPaths && ! isString( styleValue ) ) {
-				Object.entries( subPaths ).forEach( ( entry ) => {
-					const [ name, subPath ] = entry;
-					const value = get( styleValue, [ subPath ] );
-
-					if ( value ) {
-						output[ name ] = compileStyleValue( value );
-					}
-				} );
-			} else if ( ! ignoredStyles.includes( path.join( '.' ) ) ) {
-				output[ propKey ] = compileStyleValue( get( styles, path ) );
+			if ( ! styleValue.useEngine ) {
+				if ( !! subPaths && ! isString( styleValue ) ) {
+					Object.entries( subPaths ).forEach( ( entry ) => {
+						const [ name, subPath ] = entry;
+						const value = get( styleValue, [ subPath ] );
+
+						if ( value ) {
+							output[ name ] = compileStyleValue( value );
+						}
+					} );
+				} else if ( ! ignoredStyles.includes( path.join( '.' ) ) ) {
+					output[ propKey ] = compileStyleValue(
+						get( styles, path )
+					);
+				}
 			}
 		}
 	} );
 
+	// The goal is to move everything to server side generated engine styles
+	// This is temporary as we absorb more and more styles into the engine.
+	const extraRules = getCSSRules( styles, 'self' );
```

Comment:
Will `getCSSRules` need to return React-safe style camel case props as well, e.g., `{ style: { paddingTop: '19px', ... } }

I'm seeing mixed output:

<img width="500" alt="Screen Shot 2022-01-17 at 12 41 15 pm" src="https://user-images.githubusercontent.com/6458278/149688050-ca129e13-d650-4d50-b675-d56e9fd052cf.png">

I know the focus is on rendering styles to the page, so apologies if I'm jumping the 🔫

Reply:
Indeed, I guess I didn't pay enough attention here :).

For me `getCSSRules` was supposed to be an internal API (not exposed) but I had to find something to make the current style hook and theme.json style renderer continue to work with the padding moved to the style engine. So I just used what I had. I'll need to check where the mixing happens and just pick one for now.

Reply:
Interesting problem here. I had a play with a couple of different approaches and settled on switching `getCSSRules` to return the camelCase keys (since it's an array of JS objects, it felt slightly more natural to me for it to be using the JS style attribute convention), and then get `generate` to be responsible for converting it to kebab-case when outputting the CSS styles. Updated in this commit: https://github.com/WordPress/gutenberg/pull/37978/commits/a756e443a95da01dc20493099a96668ee8b20ab9

That seemed to fix it for me, though it does introduce importing another couple of `lodash` functions. I imagine we might want to see if we can get rid of the `lodash` dependency if we're expecting the style engine to also be run in standalone frontend environments outside of Gutenberg, but for the moment at least, it made it quicker to get the fix in.

Happy to change the behaviour here if folks have a preference for the shape of the attributes!